enforcer/datapath/datapath_tcp.go

```diff
@@ -529,14 +500,14 @@ func (d *Datapath) processNetworkSynPacket(context *pucontext.PUContext, conn *c
 	claims.T.AppendKeyValue(enforcerconstants.PortNumberLabelString, strconv.Itoa(int(tcpPacket.DestinationPort)))
 
 	// Validate against reject rules first - We always process reject with higher priority
-	if index, plc := context.RejectRcvRules.Search(claims.T); index >= 0 {
+	if index, plc := context.SearchRejectRcvRules(claims.T); index >= 0 {
 		// Reject the connection
-		d.reportRejectedFlow(tcpPacket, conn, txLabel, context.ManagementID, context, collector.PolicyDrop, plc.(*policy.FlowPolicy))
-		return nil, nil, fmt.Errorf("connection rejected because of policy: %+v", claims.T)
+		d.reportRejectedFlow(tcpPacket, conn, txLabel, context.ManagementID(), context, collector.PolicyDrop, plc.(*policy.FlowPolicy))
+		return nil, nil, fmt.Errorf("Connection rejected because of policy %+v", claims.T)
```

Comment:
Can you check if its better to use %+v or strings.Join() ? Use whichever you find is better and let me know too :-)